src/legacy/core_plugins/metrics/public/index.ts

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+// @ts-ignore
+import { functionsRegistry } from 'plugins/interpreter/registries';
+// @ts-ignore
+import { tsvb } from './tsvb_fn';
+// @ts-ignore
+import { MetricsVis } from './kbn_vis_type';
+import { visualizations } from '../../visualizations/public';
+
+export class Plugin {
+  // @ts-ignore
+  public setup(core, plugins) {
+    plugins.data.expressions.functionsRegistry.register(tsvb);
+    // register the provider with the visTypes registry so that other know it exists
+    plugins.visualizations.types.VisTypesRegistryProvider.register(() => MetricsVis);
+  }
+
+  public start() {}
+
+  public stop() {}
+}
+
+const core = {};
+const plugins = {
+  data: {
+    expressions: {
+      functionsRegistry,
+    },
+  },
+  visualizations,
+};
+
+new Plugin().setup(core, plugins);
```

Comment:
Perhaps for consistency we should move lines 41-51 (along with their corresponding imports) to another file `setup.ts`, as we've done in `core_plugins/data/public/setup.ts`. Then call that setup file from our hack.

This has the benefit of cleanly separating the "shim" code from the plugin definition, so the plugin definition can be a "clean" representation of how it will look once we are fully migrated to the new platform.

you could also consider moving the `Plugin` class itself to a dedicated `plugin.ts` file -- I'm currently working on doing this for the data plugin here: https://github.com/elastic/kibana/pull/39554

Reply:
Make sense. Done.